policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -819,9 +846,45 @@ impl ServiceRoutes {
     }
 
     fn delete(&mut self, gknn: &GroupKindNamespaceName) {
-        for watch in self.watches_by_ns.values_mut() {
-            watch.routes.remove(gknn);
-            watch.send_if_modified();
+        self.watches_by_ns.values_mut().for_each(|watch| {
```

Comment:
motivation for this style change?

Reply:
Same as [the other](https://github.com/linkerd/linkerd2/pull/12664#discussion_r1626126329) `for_each` change. Reverted and resolved.